### PR TITLE
Allow resizing of underlying ArrayBuffer from Rust

### PR DESCRIPTION
**Pull request by hansl**
I have an API that might truncate an `ArrayBuffer` passed in (or make it bigger). I have to call `resize()` from Rust for a buffer passed by JavaScript, and this is the best way I found to do that.

### CODE REVIEW

**Comment by jedel1043** [comment]
Does it make sense to merge this after the `usize` change gets merged? I think it would be easier to rebase this than to rebase the other PR which is a lot bigger.

**Comment by hansl** [comment]
Sure I'll wait for that PR to be merged then rebase this one.

**Comment by hansl** [comment]
@jedel1043 Please review and ready for merge :) The other PR is back to draft for the moment.

**Review by jedel1043 — COMMENTED** [review]
Why not call `resize` from here? It would remove a lot of duplication.

File: core/engine/src/builtins/array_buffer/mod.rs
Comment (jedel1043) [inline core/engine/src/builtins/array_buffer/mod.rs:588]: Why not call `resize` from here? It would remove a lot of duplication.

Comment (hansl) [reply core/engine/src/builtins/array_buffer/mod.rs:588]: I didn't want to change the behaviour of JS code.

**Review by jedel1043 — APPROVED** [review]
Looks good!

**Review by nekevss — APPROVED** [review]

**Pull request merged**
